auth/src/main/java/com/farsitel/bazaar/storage/connection/ServiceStorageConnection.kt

```diff
@@ -0,0 +1,181 @@
+                if (connect) {
+                    connectionThreadSecure = AbortableCountDownLatch(1)
+                }
+                connect
+            } catch (e: SecurityException) {
+                false
+            }
+        } ?: false
+    }
+
+    override fun getSavedData(owner: LifecycleOwner?, callback: BazaarStorageCallback) {
+        withServiceInBackground(
+            func = {
+                val bundle = getSavedData(it)
+                runOnMainThread {
+                    callback.onDataReceived(getGetSavedDataResponse(bundle))
+                }
+            },
+            onException = {
+                runOnMainThread {
+                    callback.onDataReceived(getGetSavedDataResponse(extras = null))
+                }
+            })
+    }
+
+    override fun getSavedDataSync(owner: LifecycleOwner?): BazaarResponse<ByteArray> {
+        val bundle = withService(
+            func = {
+                getSavedData(it)
+            },
+            onException = {
+                null
+            }
+        )
+
+        return getGetSavedDataResponse(bundle)
+    }
+
+    private fun getSavedData(connection: InAppBazaarStorage): Bundle? {
+        return connection.getSavedData(context.packageName)
+    }
+
+    override fun saveData(
+        owner: LifecycleOwner?,
+        data: ByteArray,
+        callback: BazaarStorageCallback
+    ) {
+        withServiceInBackground(
+            func = {
+                val bundle = saveData(it, data)
+                val bazaarResponse = getSetDataResponse(bundle)
+                runOnMainThread {
+                    callback.onDataReceived(bazaarResponse)
+                }
+            },
+            onException = {
+                runOnMainThread {
+                    callback.onDataReceived(getSetDataResponse(null))
+                }
+            }
+        )
+        saveDataSync(owner, data)
+    }
+
+    override fun saveDataSync(owner: LifecycleOwner?, data: ByteArray) {
+        withService(
+            func = {
+                saveData(it, data)
+            },
+            onException = {
+                null
+            }
+        )
+    }
+
+    private fun saveData(connection: InAppBazaarStorage, data: ByteArray): Bundle? {
+        return connection.saveData(context.packageName, data.toBase64())
+    }
+
+    override fun disconnect(context: Context) {
+        context.unbindService(this)
+        storageService = null
+        connectionThreadSecure?.abort()
+
+        backgroundThread.interrupt()
+        super.disconnect(context)
+    }
+
+    override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+        storageService = InAppBazaarStorage.Stub.asInterface(service)
+        connectionThreadSecure?.countDown()
+    }
+
+    override fun onServiceDisconnected(name: ComponentName?) {
+        storageService = null
+    }
+
+    private fun <T> withService(
```

Comment:
Same Here (I dont know if we can handle InAppBazaarStorage and Auth in the same way

Reply:
Done